packages/remix-dev/config.ts

```diff
@@ -359,6 +359,10 @@ export async function readConfig(
     );
   }
   let devServerPort = await getPort({ port: appConfig.devServerPort });
+  // check if the port is already in use (e.g this file is reloaded after a dep install or config change)
+  if (process.env.REMIX_DEV_SERVER_WS_PORT) {
+    devServerPort = Number(process.env.REMIX_DEV_SERVER_WS_PORT || "8002");
+  }
```

Comment:
Doing things this way avoids the unnecessary `getPort` call:

```suggestion
  let devServerPort;
  // check if the port is already in use (e.g this file is reloaded after a dep install or config change)
  if (process.env.REMIX_DEV_SERVER_WS_PORT) {
    devServerPort = Number(process.env.REMIX_DEV_SERVER_WS_PORT || "8002");
  } else {
    devServerPort = await getPort({ port: appConfig.devServerPort })
  }
```

Reply:
Is the `|| "8002"` necessary? `REMIX_DEV_SERVER_WS_PORT` has already been determined to be truthy.